packages/extension-ui/src/components/Toast/Toast.tsx

```diff
@@ -1,36 +1,96 @@
 // Copyright 2019-2023 @polkadot/extension-ui authors & contributors
 // SPDX-License-Identifier: Apache-2.0
 
-import type { ThemeProps } from '../../types';
+import type { SnackbarTypes, ThemeProps } from '../../types';
 
-import React from 'react';
+import React, { useCallback } from 'react';
 import styled from 'styled-components';
 
-interface Props {
+import successIcon from '../../assets/checkmark.svg';
+import criticalIcon from '../../assets/critical.svg';
+import infoIcon from '../../assets/information.svg';
+import warningIcon from '../../assets/warning.svg';
+import ToastCloseIcon from './ToastCloseIcon';
+import { TOAST_TIMEOUT } from './ToastProvider';
+
+interface Props extends ThemeProps {
   content: React.ReactNode;
   className?: string;
+  type: SnackbarTypes;
 }
 
-function Toast({ className, content }: Props): React.ReactElement<Props> {
+function Toast({ className, content, type }: Props): React.ReactElement<Props> {
+  const _getIconByType = useCallback((type: SnackbarTypes): string => {
+    switch (type) {
+      case 'success':
+        return successIcon;
+      case 'info':
+        return infoIcon;
+      case 'warning':
+        return warningIcon;
+      case 'critical':
+        return criticalIcon;
+      default:
+        return infoIcon;
```

Comment:
Just as a fun idea, up to you to decide.
You can create an icons list `.ts` file where you would reexport files like 

```js
export { default as success } from '../assets/checkmark.svg';
export { default as critical } from '../assets/critical.svg';
```

and then you could just do 

```typescript
import * as icons from './iconsList'
const _getIconByType = useCallback((type: SnackbarTypes): string => icons?.[type] ?? icons.info)

```